koku/api/provider/models.py

```diff
@@ -197,12 +197,187 @@ def delete(self, *args, **kwargs):
             using = router.db_for_write(self.__class__, isinstance=self)
             with schema_context(self.customer.schema_name):
                 LOG.info(f"PROVIDER {self.name} ({self.pk}) CASCADE DELETE -- SCHEMA {self.customer.schema_name}")
-                cascade_delete(self.__class__, self.__class__.objects.filter(pk=self.pk))
+                with transaction.atomic():
+                    self._cascade_delete()
+                    self._delete_from_target(
```

Comment:
At this point, the only thing left is the Provider, right? Can we just call `super().delete()`?

Reply:
The point here is to not call `super.delete()` unless we _have_ to.
Django ORM `delete()` calls will walk down all of the relations and try to get the record into an object or create a new query and then call the `delete()` method there. It's a mess!

Reply:
After we've called `self._cascade_delete()`, what is left? Is it not just the provider at this point?